File: src/main/java/gargoyle/calendar/util/config/Configuration.java
package gargoyle.calendar.util.config;

import gargoyle.calendar.util.convert.DefaultConverters;

import java.util.Collections;
import java.util.Map;


public final class Configuration {
    private final Map<String, String> map;

    Configuration(Map<String, String> map) {
        this.map = Collections.unmodifiableMap(map);
    }

    public String get(String key) {
        return map.get(key);
    }

    @Override
    public String toString() {
        return String.format("Configuration{linkedHashMap=%s}", map);
    }


    public <T> T getAs(String key, Class<T> type) {
        String value = get(key);
        return DefaultConverters.INSTANCE.convert(type, value);
    }

    public String getOrDefault(String key, String defaultValue) {
        String v = get(key);
        return v != null ? v : defaultValue;
    }
}

File: cal-core/src/main/kotlin/gargoyle/calendar/util/Configuration.kt
package gargoyle.calendar.util

class Configuration internal constructor(private val map: Map<String, String>) {

    operator fun get(key: String): String? = map[key]

    fun getOrDefault(key: String, defaultValue: String): String = get(key) ?: defaultValue

    override fun toString(): String = "Configuration{linkedHashMap=$map}"

}
